data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/cnec/CsaProfileCnecCreator.java

```diff
@@ -318,6 +359,50 @@ private boolean addVoltageLimit(String assessedElementId, VoltageCnecAdder volta
         return this.addVoltageLimitThreshold(assessedElementId, voltageCnecAdder, cnecLimit);
     }
 
+    private boolean addAngleLimit(String assessedElementId, AngleCnecAdder angleCnecAdder, boolean inBaseCase) {
+        boolean isErProfileDataCheckOk = this.erProfileDataCheck(assessedElementId, cnecLimit);
+
+        if (!isErProfileDataCheckOk) {
+            return false;
+        }
+
+        String isFlowToRefTerminalStr = cnecLimit.get(CsaProfileConstants.REQUEST_IS_FLOW_TO_REF_TERMINAL);
+        boolean isFlowToRefTerminalIsNull = isFlowToRefTerminalStr == null;
+        boolean isFlowToRefTerminal = isFlowToRefTerminalIsNull || Boolean.parseBoolean(isFlowToRefTerminalStr);
+
+        String terminal1Id = cnecLimit.getId("terminal1"); // "terminal1" ?
+        String terminal2Id = cnecLimit.getId("terminal2"); // "terminal2" ?
+
+        String networkElement1Id = checkNetworkElementAndGetId(assessedElementId, terminal1Id);
+        if (networkElement1Id == null) {
+            return false;
+        }
+        String networkElement2Id = checkNetworkElementAndGetId(assessedElementId, terminal2Id);
+        if (networkElement2Id == null) {
+            return false;
+        }
+
+        boolean areNetworkElementsOk = this.addAngleCnecElements(assessedElementId, angleCnecAdder, networkElement1Id, networkElement2Id, isFlowToRefTerminal);
+        if (!areNetworkElementsOk) {
+            return false;
+        }
+        this.addAngleLimitInstant(angleCnecAdder, inBaseCase);
+        return this.addAngleLimitThreshold(assessedElementId, angleCnecAdder, cnecLimit, isFlowToRefTerminalIsNull);
+    }
+
+    private String checkNetworkElementAndGetId(String assessedElementId, String terminalId) {
+        Identifiable<?> networkElement = this.getNetworkElementInNetwork(terminalId);
+        if (networkElement == null) {
+            csaProfileCnecCreationContexts.add(CsaProfileCnecCreationContext.notImported(assessedElementId, ImportStatus.ELEMENT_NOT_FOUND_IN_NETWORK, "angle limit equipment is missing in network : " + terminalId));
+            return null;
+        }
+        if (!(networkElement instanceof BusbarSection)) {
```

Comment:
BusbarSection and Bus are not the same
```suggestion
        if (!networkElement.getType().equals(IdentifiableType.BUS)) {
```